src/ngraph/op/fused/squeeze.cpp

```diff
@@ -32,6 +32,17 @@ op::Squeeze::Squeeze(const Output<Node>& data, const Output<Node>& axes)
     constructor_validate_and_infer_types();
 }
 
+void op::Squeeze::pre_validate_and_infer_types()
+{
+    auto axes_node = input_value(1).get_node_shared_ptr();
+    // Currently only support Constant node for axes.
+    NODE_VALIDATION_CHECK(this,
+                          axes_node->is_constant(),
+                          "doesn't support 'axes' input of other type than a Constant.");
+
+    set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
+}
+
```

Comment:
it might be better to take the output shape computation logic from decompose op and override `validate_and_infer_types` instead of restricting this op to only constant reduction_axes input.

Reply:
Pushed changes. Is that what you requested?